File: app/src/main/java/igrek/todotree/activity/QuickAddActivity.java
package igrek.todotree.activity;

import android.content.Intent;
import android.os.Bundle;

public class QuickAddActivity extends MainActivity {
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		quickAddService.enableQuickAdd();
	}
	
	@Override
	protected void onNewIntent(Intent intent) {
		stdNewIntent(intent);
		logger.debug("recreating new quick add activity");
		recreate();
	}
	
}

File: app/src/main/java/igrek/todotree/activity/QuickAddActivity.kt
package igrek.todotree.activity

import android.content.Intent
import android.os.Bundle

class QuickAddActivity : MainActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityData.quickAddService.enableQuickAdd()
    }

    override fun onNewIntent(intent: Intent?) {
        stdNewIntent(intent)
        logger.debug("recreating new quick add activity")
        recreate()
        null?.let {
            super.onNewIntent(intent)
        }
    }
}